processing/deps/rasters_to_netcdf.py

```diff
@@ -0,0 +1,182 @@
+
+    base_geotransform = datasets[0].GetGeoTransform()
+
+    for dataset in datasets[1:]:
+        current_geotransform = dataset.GetGeoTransform()
+        if base_geotransform != current_geotransform:
+            return False
+    return True
+
+
+def rasters_have_same_dimensions(datasets: List[gdal.Dataset]):
+    if not datasets:
+        raise ValueError("The filepaths list is empty.")
+
+    base_dimensions = datasets[0].RasterXSize, datasets[0].RasterYSize
+
+    for dataset in datasets[1:]:
+        current_dimensions = dataset.RasterXSize, dataset.RasterYSize
+        if base_dimensions != current_dimensions:
+            return False
+    return True
+
+
+def rasters_have_same_nodatavalue(datasets: List[gdal.Dataset]):
+    if not datasets:
+        raise ValueError("The filepaths list is empty.")
+
+    base_ndv = datasets[0].GetRasterBand(1).GetNoDataValue()
+
+    for dataset in datasets[1:]:
+        current_ndv = dataset.GetRasterBand(1).GetNoDataValue()
+        if base_ndv != current_ndv:
+            return False
+    return True
+
+
+def rasters_to_netcdf(
+        rasters: List[Union[str, Path]],
+        start_time: datetime,
+        interval: int,
+        units: str,
+        output_path: Union[str, Path],
+        time_units: str = 'seconds since 1970-01-01 00:00:00.0 +0000',
+        calendar: str = 'standard',
+        offset: int = 0
+) -> None:
+    """
+    :param interval: interval in seconds
+    :param units: one of 'mm', 'm/s', 'mm/h', 'mm/hr'. Note: in case of `mm` the rate is determined by looking at the
+       next `time` value.
+    :param offset: offset in seconds
+    """
+    datasets = get_datasets(rasters)
+    assert rasters_have_same_srs(datasets), "Not all input rasters have the same Spatial Reference System"
+    assert rasters_have_same_geotransform(datasets), "Not all input rasters have the same origin, pixel size, and skew"
+    assert rasters_have_same_dimensions(datasets), "Not all input rasters have the same width and height"
+    assert rasters_have_same_nodatavalue(datasets), "Not all input rasters have the same nodatavalue"
+
+    srs = get_srs(datasets[0])
+    crs = CRS.from_wkt(srs.ExportToWkt())
+    geotransform = datasets[0].GetGeoTransform()
+
+    # create netcdf
+    output_dataset = netCDF4.Dataset(
+        filename=str(output_path),
+        mode='w',
+        format="NETCDF4"
+    )
+
+    # dataset attributes
+    output_dataset.setncattr(name='OFFSET', value=0)
+    crs_var = output_dataset.createVariable(varname='crs', datatype='int')
+    crs_var.setncatts(crs.to_cf())
+    crs_var.setncattr_string(name="spatial_ref", value=crs.to_wkt())
+    crs_var.setncattr_string(name="GeoTransform", value=" ".join([str(i) for i in geotransform]))
+
+    # set dimensions
+    output_dataset.createDimension(dimname='lon', size=datasets[0].RasterXSize)
+    output_dataset.createDimension(dimname='lat', size=datasets[0].RasterYSize)
```

Comment:
in de 3Di api was er iets met x en y die soms omgedraaid zijn toch? Ik weet even niet meer of dat nu gefixt is (dus x altijd `lon` en y `lat`). If not dan zou we daar ook nog wat mee moeten doen misschien?

Reply:
dat was inderdaad zo maar dat had je eerder al gefixt